src/charm.py

```diff
@@ -583,18 +639,39 @@ def _common_exit_hook(self) -> None:
 
         self.karma_provider.target = self._external_url
 
+        # Update config file
+        self._push_tls_files()
+        try:
+            self._update_config()
+        except ConfigUpdateFailure as e:
+            self.unit.status = BlockedStatus(str(e))
+            return
+
         # Update pebble layer
         self._update_layer()
 
-        # Update config file
+        # Reload or restart the service
         try:
-            self._update_config()
+            self._reload()
         except ConfigUpdateFailure as e:
             self.unit.status = BlockedStatus(str(e))
             return
 
         self.unit.status = ActiveStatus()
 
+    def _on_server_cert_changed(self, _):
+        self._common_exit_hook()
+
+    def _push_tls_files(self):
+        if key := self.server_cert.key:
+            self.container.push(self._key_path, key, make_dirs=True)
+        else:
+            self.container.remove_path(self._key_path, recursive=True)
+        if cert := self.server_cert.cert:
+            self.container.push(self._server_cert_path, cert, make_dirs=True)
+        else:
+            self.container.remove_path(self._server_cert_path, recursive=True)
```

Comment:
Does this method name represents well what this method actually does?
What about `_update_tls_files` instead?

Reply:
Fixed in next commit.